### PR TITLE
Issue 6245 - covscan fixes

### LINKED ISSUE #6245: Fix some coverity scan issues
**Issue Description**
Fix the issue rated as "important"  reported by coverity scan static analyzer

**Package Version and Platform:**
 - Platform:  Fedora
 - Package and version: main branch 

**Additional context**

Using some internal tools to run coverity scan 
`osh-cli mock-build --config=fedora-rawhide-x86_64 $WS/dist/srpms/389-ds-base-*.src.rpm
`
But I guess this should also be visible  in the  github coverity scan test results

Triaged: https://issues.redhat.com/browse/IDMDS-4302

### PR DESCRIPTION
**Pull request by progier389**
Fix issues reported by coverity scan static analyzer

Issue: #6245 

Reviewed by: @mreynolds389 (Thanks!)

### CODE REVIEW

File: ldap/servers/slapd/back-ldbm/ldbm_search.c [hunk 1]
```diff
@@ -1929,13 +1929,6 @@ ldbm_back_next_search_entry(Slapi_PBlock *pb)
             }
         }
     }
-    /* check for the final abandon */
-    if (slapi_op_abandoned(pb)) {
-        slapi_pblock_set(pb, SLAPI_SEARCH_RESULT_SET_SIZE_ESTIMATE, &estimate);
-        slapi_pblock_set(pb, SLAPI_SEARCH_RESULT_ENTRY, NULL);
-        delete_search_result_set(pb, &sr);
-        rc = SLAPI_FAIL_GENERAL;
-    }
```
Comment (mreynolds389) [inline ldap/servers/slapd/back-ldbm/ldbm_search.c:1938]: What coverity issue is this fixing?  Just curious :-)

File: src/lib389/lib389/cli_ctl/dsrc.py [hunk 2]
```diff
@@ -199,8 +199,9 @@ def modify_dsrc(inst, log, args):
         if args.starttls:
              config[inst.serverid]['starttls'] = 'true'
         if args.cancel_starttls:
-            # coverity[copy_paste_error]
-            config[inst.serverid]['starttls'] = 'false'
+            # coverity[copy_paste_error] does not work as expecpted
```
Comment (mreynolds389) [inline src/lib389/lib389/cli_ctl/dsrc.py]: Typo, "expected", the rest looks good to me!

**Review by mreynolds389 — APPROVED** [review]

File: ldap/servers/slapd/back-ldbm/ldbm_search.c (hunk 1 above)
Comment (progier389) [reply ldap/servers/slapd/back-ldbm/ldbm_search.c:1938]: Unreachable code. 
There is no break in the while(1) loop but only some "goto bail;"
so the code between the end of the loop and the bail label is unreachable

**Comment by progier389** [comment]
Fixed the comment typo.

**Pull request merged**